### PR TITLE
Implement lua rng

### PR DESCRIPTION
**Pull request by nullkiller**
VR:T and VR:R

- [x] Need to seed VR:R somehow

### CODE REVIEW

File: scripts/lib/erm/VR.lua [hunk 1]
```diff
@@ -74,5 +74,18 @@ function VR:M6()
...
 
+function VR:T(maxVal)
+	local rng = Rng.getDefault()
+	local r = rng:nextInt(maxVal)
+	
+	return self.v + r
+end
+
+function VR:R(maxVal)
+	local rng = Rng.getDefault()
```
Comment (nullkiller) [inline scripts/lib/erm/VR.lua:85]: We also need a way to get rng with some fixed initial seed or rng from gamestate for VR:R. @alexvins, any ideas?

Comment (alexvins) [reply scripts/lib/erm/VR.lua:85]: We should extend ServerCbProxy API with `ServerCallback::getRNG`.

Comment (nullkiller) [reply scripts/lib/erm/VR.lua:85]: Thank you. Will try to do it.

Comment (nullkiller) [reply scripts/lib/erm/VR.lua]: Fixed this.

File: scripting/lua/api/ServerCb.cpp [hunk 2]
```diff
@@ -43,9 +45,27 @@ const std::vector<ServerCbProxy::CustomRegType> ServerCbProxy::REGISTER_CUSTOM =
...
+int ServerCbProxy::getRNG(lua_State * L)
+{
+	LuaStack S(L);
+	ServerCallback * obj = nullptr;
+
+	if(!S.tryGet(1, obj))
+		return S.retVoid();
+
+	S.clear();
+	S.push(dynamic_cast<CRandomGenerator *>(obj->getRNG()));
```
Comment (alexvins) [inline scripting/lua/api/ServerCb.cpp:65]: it should be no cast here and API for RNG should use vstd::RNG. Also you should use `RNGMock` in tests

Comment (nullkiller) [reply scripting/lua/api/ServerCb.cpp:65]: I can try to move nextInt and nextDouble from CRandomGenerator to vstd::RNG and use vstd::RNG* in API.

Comment (alexvins) [reply scripting/lua/api/ServerCb.cpp:65]: Please do not move, just declare as pure virtual

Comment (nullkiller) [reply scripting/lua/api/ServerCb.cpp]: Fixed this. Was not able to set mocked RNG for VR:T.

**Pull request closed without merging**